consensus/merkle_proof/src/lib.rs

```diff
@@ -382,20 +382,19 @@ pub fn verify_merkle_proof(
 pub fn merkle_root_from_branch(leaf: H256, branch: &[H256], depth: usize, index: usize) -> H256 {
     assert_eq!(branch.len(), depth, "proof length should equal depth");
 
-    let mut merkle_root = leaf.as_slice().to_vec();
+    let mut merkle_root = leaf.0;
```

Comment:
Output of the hash function is a H256/[u8;32] so we can operate on [u8;32] internally and skip the heap allocations on each loop